src/pkgcore/ebuild/repo_objs.py

```diff
@@ -170,6 +171,217 @@ def __init__(self, metadata_xml, manifest):
         self.manifest = manifest
 
 
+class ProjectMember(object, metaclass=klass.generic_equality):
+    """Data on a single project member.
+
+    :type email: C{unicode} object
+    :ivar email: email address.
+    :type name: C{unicode} object or C{None}
+    :ivar name: full name
+    :type role: C{unicode} object or C{None}
+    :ivar role: role within the project.
+    :type is_lead: C{bool}
+    :ivar is_lead: whether the member is a project lead.
+    """
+
+    __slots__ = ('email', 'name', 'role', 'is_lead')
+    __attr_comparison__ = ('email', 'name', 'role', 'is_lead')
+
+    def __init__(self, email, name=None, role=None, is_lead=None):
+        if email is None:
+            raise ValueError('email for project member must not be null')
+        self.email = email
+        self.name = name
+        self.role = role
+        self.is_lead = is_lead
+
+    def __str__(self):
+        if self.name is not None:
+            res = f'{self.name} <{self.email}>'
+        else:
+            res = self.email
+        if self.role is not None:
+            return f'{res} ({self.role})'
+        return res
+
+
+class Subproject(object):
+    """Data on a subproject.
+
+    :type inherit_members: C{bool}
+    :ivar inherit_members: whether the parent project inherits members
+                           from this subproject.
+    """
+
+    __slots__ = ('_ref', 'inherit_members', '_projects_xml',
+                 '_cached_project')
+
+    def __init__(self, ref, projects_xml, inherit_members=None):
+        if ref is None:
+            raise ValueError('ref for subproject must not be null')
+        self._ref = ref
+        self.inherit_members = inherit_members
+        self._projects_xml = projects_xml
+        self._cached_project = None
+
+    @property
+    def _project(self):
+        if self._cached_project is None:
+            self._cached_project = self._projects_xml.projects[self._ref]
+        return self._cached_project
+
+    __getattr__ = klass.GetAttrProxy("_project")
+    # TODO: radhermit, why doesn't this work? because of __slots__?
```

Comment:
Possibly because of __slots__, yes. Make _project a jitted attr, make sure to add it's underlying attr name to __slots__ (jitted attrs by default cache their values at '_' + func_name) and it should work.

Reply:
I managed to make jitattr work but this one still fails.

Reply:
this one = dirproxy